extensions/amp-stream-gallery/1.0/stream-gallery.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {
+  BaseCarousel,
+  Controls,
+} from '../../amp-base-carousel/1.0/base-carousel';
+import {CarouselContext} from '../../amp-base-carousel/1.0/carousel-context';
+import {ContainWrapper} from '../../../src/preact/component';
+import {useCallback, useMemo, useRef, useState} from '../../../src/preact';
+import {useStyles} from './stream-gallery.jss';
+
+/**
+ * @param {!StreamGalleryDef.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function StreamGallery({
+  arrowPrev: customArrowPrev,
+  arrowNext: customArrowNext,
+  children,
+  extraSpace,
+  insetArrowVisibility,
+  loop,
+  maxItemWidth = Number.MAX_VALUE,
+  minItemWidth = 1,
+  maxVisibleCount = Number.MAX_VALUE,
+  minVisibleCount = 1,
+  outsetArrows,
+  peek = 0,
+  slideAlign,
+  snap,
+  style,
+  ...rest
+}) {
+  const [currentSlide, setCurrentSlide] = useState(0);
+  const [slideCount, setSlideCount] = useState(null);
+  const carouselContext = useMemo(
+    () => ({currentSlide, setCurrentSlide, slideCount, setSlideCount}),
+    [currentSlide, slideCount]
+  );
+  const galleryRef = useRef(null);
+  const carouselRef = useRef(null);
+  const advance = useCallback((by) => carouselRef.current.advance(by), []);
+  const {visibleCount, maxContainerWidth} = getVisibleCount(
+    maxItemWidth,
+    minItemWidth,
+    maxVisibleCount,
+    minVisibleCount,
+    children.length,
+    peek,
+    galleryRef
+  );
+  const arrowPrev = customArrowPrev || (
+    <DefaultArrow
+      advance={advance}
+      by={-visibleCount}
+      outsetArrows={outsetArrows}
+      setCurrentSlide={setCurrentSlide}
+    />
+  );
+  const arrowNext = customArrowNext || (
+    <DefaultArrow
+      advance={advance}
+      by={visibleCount}
+      outsetArrows={outsetArrows}
+      setCurrentSlide={setCurrentSlide}
+    />
+  );
+
+  return (
+    <ContainWrapper
+      layout={true}
+      size={true}
```

Comment:
`paint={true}` is also probably a good idea?

Reply:
Removed `ContainWrapper` entirely.